connectors/byoc.py

```diff
@@ -668,71 +616,99 @@ async def prepare_docs(self, data_provider, pipeline=None, filtering=None):
                 logger.error(
                     f"Service type is not configured for connector {configured_connector_id}"
                 )
                 raise ServiceTypeNotConfiguredError("Service type is not configured.")
-            self.service_type = configured_service_type
-            logger.debug(f"Populated service type for connector {self.id}")
-
-        service_type = self.service_type
-        if service_type not in config["sources"]:
-            raise ServiceTypeNotSupportedError(service_type)
+            self.service_type = doc["service_type"] = configured_service_type
+            logger.debug(
+                f"Populated service type {configured_service_type} for connector {self.id}"
+            )
 
-        fqn = config["sources"][service_type]
-        try:
-            source_klass = get_source_klass(fqn)
+        if self.configuration.is_empty():
+            if configured_service_type not in config["sources"]:
+                raise ServiceTypeNotSupportedError(configured_service_type)
+            fqn = config["sources"][configured_service_type]
+            try:
+                source_klass = get_source_klass(fqn)
 
-            if self.configuration.is_empty():
                 # sets the defaults and the flag to NEEDS_CONFIGURATION
-                self.doc_source[
+                self.configuration = doc[
                     "configuration"
                 ] = source_klass.get_simple_configuration()
-                self.doc_source["status"] = Status.NEEDS_CONFIGURATION.value
-                self._update_config(self.doc_source)
+                self.status = doc["status"] = Status.NEEDS_CONFIGURATION.value
                 logger.debug(f"Populated configuration for connector {self.id}")
-
-            # sync state if needed (when service type or configuration is updated)
-            try:
-                await self.sync_doc(force=False)
             except Exception as e:
                 logger.critical(e, exc_info=True)
-                raise ConnectorUpdateError(
-                    f"Could not update configuration for connector {self.id}"
+                raise DataSourceError(
+                    f"Could not instantiate {fqn} for {configured_service_type}"
                 )
+
+        try:
+            await self.index.update(doc_id=self.id, doc=doc)
         except Exception as e:
             logger.critical(e, exc_info=True)
-            raise DataSourceError(f"Could not instantiate {fqn} for {service_type}")
+            raise ConnectorUpdateError(
+                f"Could not update service type/configuration for connector {self.id}"
+            )
+
+    async def validate_filtering(self, validator):
+        draft_filter = self.filtering.get_draft_filter()
+        if not draft_filter.has_validation_state(FilteringValidationState.EDITED):
+            return
 
-        self.source_klass = source_klass
+        logger.info(f"Filtering of connector {self.id} is edited, validating...)")
```

Comment:
```suggestion
        logger.info(f"Filtering of connector {self.id} is {FilteringValidationState.EDITED.value}, validating...)")
```

Just to be more robust against possible changes.